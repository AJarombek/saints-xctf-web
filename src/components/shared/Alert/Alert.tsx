/**
 * Component for an Alert that pops up with warnings or errors.
 * @author Andrew Jarombek
 * @since 1/14/2021
 */

import React, { ReactNode } from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';

export type AlertType = 'error' | 'warning' | 'info' | 'success';

interface Props {
  message: ReactNode;
  type: AlertType;
  closeable: boolean;
  onClose?: () => void;
}

const useStyles = createUseStyles(styles);

const Alert: React.FunctionComponent<Props> = ({ message, type, closeable, onClose }) => {
  const classes = useStyles({ type });

  let alertIcon;
  switch (type) {
    case 'error':
      alertIcon = '\ue062';
      break;
    case 'info':
      alertIcon = '\ue064';
      break;
    case 'warning':
      alertIcon = '\ue063';
      break;
    case 'success':
      alertIcon = '\ue052';
      break;
    default:
      alertIcon = '\ue062';
  }

  return (
    <div className={classes.alert} data-cypress="alert">
      <p className={classes.alertIcon}>{alertIcon}</p>
      <div className={classes.message}>{message}</div>
      {closeable && (
        <p className={classes.closeIcon} onClick={onClose}>
          &#x4d;
        </p>
      )}
    </div>
  );
};

export default Alert;
