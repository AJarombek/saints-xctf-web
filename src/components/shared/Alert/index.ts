import Alert, { AlertType } from './Alert';

export { AlertType };
export default Alert;
