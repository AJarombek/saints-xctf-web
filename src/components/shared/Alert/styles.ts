/**
 * JSS styles for the Alert component.
 * @author Andrew Jarombek
 * @since 1/14/2021
 */

import { AlertType } from './Alert';
import color from 'color';
import Colors from '../../../styles/colors';
import { FontMixins } from '../../../styles/mixins';

export default {
  alert: {
    display: 'flex',
    alignItems: 'center',
    padding: '10px 0',
    borderRadius: 3,
    backgroundColor: ({ type }: { type: AlertType }): string =>
      type === 'warning'
        ? color(Colors.statusWarning).lighten(0.65).hex()
        : type === 'info'
        ? color(Colors.spotPaletteBlue).lighten(0.65).hex()
        : color(Colors.statusFailure).lighten(0.65).hex()
  },
  alertIcon: {
    ...FontMixins.elegantIcons(),
    fontSize: 28,
    margin: '10px 25px',
    color: ({ type }: { type: AlertType }): string =>
      type === 'warning' ? Colors.statusWarning : type === 'info' ? Colors.spotPaletteBlue : Colors.statusFailure
  },
  message: {
    ...FontMixins.roboto(),
    fontSize: 16,
    margin: '5px 10px 5px 0'
  },
  closeIcon: {
    ...FontMixins.elegantIcons(),
    fontSize: 24,
    margin: '0 10px 0 auto',
    cursor: 'pointer'
  },
  '@media screen and (max-width: 1000px)': {
    alertIcon: {
      margin: 10
    },
    message: {
      margin: 0
    }
  }
};
