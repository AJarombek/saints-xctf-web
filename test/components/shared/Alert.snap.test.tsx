/**
 * Snapshot test for the Alert component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Alert from '../../../src/components/shared/Alert';

describe('Alert Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Alert message="An error occurred." type="error" closeable={true} onClose={(): void => null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
