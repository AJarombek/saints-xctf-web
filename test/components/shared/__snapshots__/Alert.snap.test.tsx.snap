// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshot Tests renders correctly 1`] = `
<div
  className="alert-0-1-1 alert-d0-0-1-5"
  data-cypress="alert"
>
  <p
    className="alertIcon-0-1-2 alertIcon-d1-0-1-6"
  >
    ÓÅ¢
  </p>
  <div
    className="message-0-1-3"
  >
    An error occurred.
  </div>
  <p
    className="closeIcon-0-1-4"
    onClick={[Function]}
  >
    M
  </p>
</div>
`;
